File: iOS/Utils/Utils/Utils/ConfigUtils.m
#import "ConfigUtils.h"
#import "ReflectionUtils.h"

@implementation ConfigUtils

+(NSString *) loadConfigString: (NSString *) key {
    return (NSString *)[self loadConfigObj:key];
}

+(int) loadConfigInt: (NSString *) key {
    return [(NSNumber *)[self loadConfigObj:key] intValue];
}

+(NSNumber *) loadConfigNumber: (NSString *) key {
    return (NSNumber *)[self loadConfigObj:key];
}

+(BOOL) loadConfigBool: (NSString *) key {
    return [(NSNumber *)[self loadConfigObj:key] boolValue];
}

+(id) loadConfigObj: (NSString *) key {
    return [[NSUserDefaults standardUserDefaults] objectForKey:key];
}

+(void) saveConfigString: (NSString *) key value: (NSString *) val {
    [self saveConfigObj:key value:val];
}

+(void) saveConfigInt: (NSString *) key value: (int) val {
    [self saveConfigObj:key value:[NSNumber numberWithInt:val]];
}

+(void) saveConfigNumber: (NSString *) key value: (NSNumber *) val {
    [self saveConfigObj:key value:val];
}

+(void) saveConfigBool: (NSString *) key value: (BOOL) val {
    [self saveConfigObj:key value:[NSNumber numberWithBool:val]];
}

+(void) saveConfigObj: (NSString *) key value: (id) val {
    NSUserDefaults * def = [NSUserDefaults standardUserDefaults];
    [def setObject:val forKey:key];
    [def synchronize];
}


+(void)loadConfigObject: (NSString *)key obj:(id)obj {
    NSMutableArray * fields = [ReflectionUtils getClassFields:obj];
    for (NSString * field in fields) {
        [obj setValue:[self loadConfigObj:[NSString stringWithFormat:@"%@_%@", key, field]] forKey:field];
    }
}

+(void)saveConfigObject: (NSString *)key obj:(id)obj {
    NSMutableArray * fields = [ReflectionUtils getClassFields:obj];
    for (NSString * field in fields) {
        [self saveConfigObj:[NSString stringWithFormat:@"%@_%@", key, field] value:[obj valueForKey:field]];
    }
}

@end

File: Swift/ConfigUtils.swift
import UIKit

let KEY_BACKGROUND = "background"

class ConfigUtils: NSObject {
    
    class func loadConfigString(key: String) -> String? {
        return loadConfigObj(key) as? String
    }
    
    class func loadConfigInt(key: String) -> Int? {
        return loadConfigObj(key) as? Int
    }
    
    class func loadConfigDouble(key: String) -> Double? {
        return loadConfigObj(key) as? Double
    }
    
    class func loadConfigBool(key: String) -> Bool? {
        return loadConfigObj(key) as? Bool
    }
    
    class func loadConfigObj(key: String) -> AnyObject? {
        return NSUserDefaults.standardUserDefaults().objectForKey(key)
    }
    
    class func saveConfigString(key: String, val: String) {
        saveConfigObj(key, val: val)
    }
    
    class func saveConfigInt(key: String, val: Int) {
        saveConfigObj(key, val: val)
    }
    
    class func saveConfigDouble(key: String, val: Double) {
        saveConfigObj(key, val: val)
    }
    
    class func saveConfigBool(key: String, val: Bool) {
        saveConfigObj(key, val: val)
    }
    
    class func saveConfigObj(key: String, val: AnyObject) {
        var def = NSUserDefaults.standardUserDefaults()
        def.setObject(val, forKey: key)
        def.synchronize()
    }
}
